CapstoneProject/Assets/Scripts/Core/BehaviorTree/AI.cs

```diff
@@ -24,10 +37,101 @@ void Update()
         
     }
 
+    bool IsHittingWall()
+    {
+        bool result;
+
+        currentFacingDirection = _rb.CurrentFacingDirection;
+        // Define the cast distance for left and right
+        float castingDistance = baseCastDistance;
+        if (currentFacingDirection == RigidbodyComponent.FacingDirections.LEFT)
+        {
+            castingDistance = -baseCastDistance;
+        }
+
+        // determine the target destination based on the cast distance
+        Vector3 targetPos = CastPos.position;
+        targetPos.x += castingDistance;
+
+        Debug.DrawLine(CastPos.position, targetPos, Color.red);
+        if (Physics2D.Linecast(CastPos.position, targetPos, 1 << LayerMask.NameToLayer("Ground")))
+        {
+            result = true;
+        }
+        else
+        {
+            result = false;
+        }
+
+        return result;
```

Comment:
I would personally just return the if contents:
return Physics2D.Linecast(CastPos.position, targetPos, 1 << LayerMask.NameToLayer("Ground");

same for the next function